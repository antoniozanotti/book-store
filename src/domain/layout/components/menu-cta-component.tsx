"use client";

import { MenuCtaComponentInterface } from "./menu-cta-component-interface";
import { MenuIcon, XIcon } from "lucide-react";
import { MenuActionEnum } from "../enums/menu-action-enum";
import { useMenuContext } from "../contexts/menu-context";

export function MenuCtaComponent({ ...props }: MenuCtaComponentInterface) {
  props.className = props.className + " flex items-center gap-2 pl-3 lg:hidden";
  const menuState = useMenuContext().state;
  const menuDispatch = useMenuContext().dispatch;

  return (
    <button
      {...props}
      onClick={() => {
        menuDispatch({ type: MenuActionEnum.CHANGED_IS_OPEN });
      }}
    >
      {menuState.isOpen ? <XIcon /> : <MenuIcon />} <span>Menu</span>
    </button>
  );
}
