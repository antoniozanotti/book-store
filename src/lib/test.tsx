import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
