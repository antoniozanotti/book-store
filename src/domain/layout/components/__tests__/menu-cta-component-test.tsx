import { render, screen } from "@/lib/test";
import { MenuCtaComponent } from "../menu-cta-component";

describe("MenuCtaComponent", () => {
  it("should render", async () => {
    const { container } = render(<MenuCtaComponent />);
    expect(container).toMatchSnapshot();
  });
});