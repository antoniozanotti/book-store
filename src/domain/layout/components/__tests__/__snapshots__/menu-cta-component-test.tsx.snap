// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCtaComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <button
      class="undefined flex items-center gap-2 pl-3 lg:hidden"
    >
      <svg
        class="lucide lucide-menu "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="20"
          y1="12"
          y2="12"
        />
        <line
          x1="4"
          x2="20"
          y1="6"
          y2="6"
        />
        <line
          x1="4"
          x2="20"
          y1="18"
          y2="18"
        />
      </svg>
       
      <span>
        Menu
      </span>
    </button>
  </div>
</div>
`;
